### PR TITLE
Update yasson to 3.0.3 and fix translated messages

### LINKED ISSUE #19698: Record objects failing silently with Jakarta EE 9.1
I'm experimenting with Java 17 and JakartaEE 9.1 restful services and running into issues with `Record` objects.

I generated a project with a simple greeting service that returns a `Record` type:

```java
    @GET
    @Path("{name}")
    @Produces(MediaType.APPLICATION_JSON)
    public GreetingResponse greeting(@PathParam("name") String name) throws WebApplicationException {
      return new GreetingResponse("Hello, " + name); 
    }
```
where `GreetingResponse` is 

```java
public record GreetingResponse(String greeting) { }
```

When accessing the resource
```sh
curl http://localhost:9080/hello-world/api/v1/greeting/Robert
```
the response is 
```javascript
{}
```

No error message is produced in the logs.

Yasson was [recently updated](https://github.com/eclipse-ee4j/yasson/releases/tag/2.0.3) to support `Record` types.  I flirted with the jsonbContainer-2.0 feature but the documentation is unclear on how to configure in server.xml or where to put the jar / whether or not a `<bell>` is required.

### PR DESCRIPTION
**Pull request by KyleAure**
Fixes #19698 

And makes sure that we include all translated messages for Yasson

### CODE REVIEW

**Review by mswatosh — DISMISSED** [review]

**Comment by KyleAure** [comment]
#build

File: dev/io.openliberty.jakarta.jsonb.internal_fat/fat/src/io/openliberty/jakarta/jsonb/internal/fat/YassonTranslationTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.jsonb.internal_fat/fat/src/io/openliberty/jakarta/jsonb/internal/fat/YassonTranslationTest.java]:
```suggestion
```

Not needed for new files.

File: dev/com.ibm.ws.jsonb_fat/test-applications/jsonbapp/src/web/jsonbtest/JSONBTestServlet.java [hunk 2]
```diff
@@ -196,7 +196,8 @@ public void testArrays() throws Exception {
             byte[] b = jsonb.fromJson(cr, byte[].class);
             assertArrayEquals(bytes, b);
 
-            cr.reset();
+            //Yasson 3.0.3 now makes sure to close parser and the stream associated with a CharArrayReader
+            cr = new CharArrayReader(cw.toCharArray());
```
Comment (jhanders34) [inline dev/com.ibm.ws.jsonb_fat/test-applications/jsonbapp/src/web/jsonbtest/JSONBTestServlet.java]:
```suggestion
            cr.close();
            cr = new CharArrayReader(cw.toCharArray());
```
When running with jsonb 1.0 and 2.0 do you want to have a cr.close()?

File: dev/io.openliberty.jakarta.jsonb.internal_fat/publish/servers/io.openliberty.jakarta.yasson.internal.fat.translation/bootstrap.properties [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# Copyright (c) 2023 IBM Corporation and others.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License 2.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-2.0/
+# 
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.jsonb.internal_fat/publish/servers/io.openliberty.jakarta.yasson.internal.fat.translation/bootstrap.properties]:
```suggestion
```
Not needed for new files.

File: dev/io.openliberty.jakarta.jsonb.internal_fat/publish/servers/io.openliberty.jakarta.yasson.internal.fat.translation/server.xml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    Copyright (c) 2023 IBM Corporation and others.
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License 2.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-2.0/
+    
+    SPDX-License-Identifier: EPL-2.0
+   
+    Contributors:
+        IBM Corporation - initial API and implementation
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.jsonb.internal_fat/publish/servers/io.openliberty.jakarta.yasson.internal.fat.translation/server.xml]:
```suggestion
```
Not needed for new files.

File: dev/io.openliberty.jakarta.jsonb.internal_fat/test-applications/yassontranslationtestapp/src/test/yasson/translation/web/YassonTranslationTestServlet.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.jsonb.internal_fat/test-applications/yassontranslationtestapp/src/test/yasson/translation/web/YassonTranslationTestServlet.java]:
```suggestion
```
Not needed in new files.

File: dev/io.openliberty.jakarta.jsonb.internal_fat/test-applications/yassontranslationtestapp/src/test/yasson/translation/web/YassonTranslationTestServlet.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    @Test
+    public void testTranslationMessageDefault() {
+        String expected = Messages.getMessage(MessageKeys.INVOKING_GETTER, Locale.getDefault());
+        String actual = Messages.getMessage(MessageKeys.INVOKING_GETTER);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testTranslationMessageNewLocale() {
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.jsonb.internal_fat/test-applications/yassontranslationtestapp/src/test/yasson/translation/web/YassonTranslationTestServlet.java]: Shouldn't this be tested with a second start of the server with the Locale set to French?

Comment (KyleAure) [reply dev/io.openliberty.jakarta.jsonb.internal_fat/test-applications/yassontranslationtestapp/src/test/yasson/translation/web/YassonTranslationTestServlet.java]: Thanks for the suggestion.  I updated this PR to include a full functional test :)

File: dev/com.ibm.ws.jsonb_fat/test-applications/jsonbapp/src/web/jsonbtest/JSONBTestServlet.java (hunk 2 above)
Comment (KyleAure) [reply dev/com.ibm.ws.jsonb_fat/test-applications/jsonbapp/src/web/jsonbtest/JSONBTestServlet.java]: Yes this would be the safe thing to do.  Updated.

**Comment by KyleAure** [comment]
#libby - recheck headers
#build - rebuild with changes

File: dev/com.ibm.ws.jsonb_fat/test-applications/jsonbapp/src/web/jsonbtest/JSONBTestServlet.java [hunk 7]
```diff
@@ -192,16 +192,24 @@ public void testArrays() throws Exception {
...
-            s = jsonb.fromJson(cr, String[].class);
-            cr.close();
-            assertEquals(bytes.length, s.length);
-            for (int i = 0; i < bytes.length; i++)
-                assertEquals("failed at position " + i, Byte.toString(bytes[i]), s[i]);
+
+            /**
+             * Yasson 3.0.3 ensures the closure of the constructed parser and consequently the Reader.
+             * Therefore, this test needed to be modified to avoid calling reset()
+             * The more pragmatic way to right this test, is to use try-with-resources to control the scope of the Reader.
```
Comment (jhanders34) [inline dev/com.ibm.ws.jsonb_fat/test-applications/jsonbapp/src/web/jsonbtest/JSONBTestServlet.java]:
```suggestion
             * The more pragmatic way to write this test, is to use try-with-resources to control the scope of the Reader.
```

File: dev/io.openliberty.jakarta.jsonb.internal_fat/fat/src/io/openliberty/jakarta/jsonb/internal/fat/YassonTranslationTest.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        jvmOptions.remove("-Duser.language");
+        jvmOptions.remove("-Duser.country");
+        server.setJvmOptions(jvmOptions);
+
+        server.startServer();
+
+        runTest(server, APP_NAME + "/" + CONTEXT, "testTranslationMessageDefault");
+        runTest(server, APP_NAME + "/" + CONTEXT, "testTranslationMessageProvidedLocale");
+
+        server.stopServer();
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.jsonb.internal_fat/fat/src/io/openliberty/jakarta/jsonb/internal/fat/YassonTranslationTest.java]: If the test fails, won't we leak the server running?  Should the server.stopServer() either be moved to a @After method using if isServerStarted() check before calling stopServer() or just have that additionally to what we are doing here.  We could also change to do a try / finally instead around the runtTest method invocations.?

File: dev/io.openliberty.jakarta.jsonb.internal_fat/fat/src/io/openliberty/jakarta/jsonb/internal/fat/YassonTranslationTest.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+import componenttest.annotation.Server;
+import componenttest.annotation.TestServlet;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import test.yasson.translation.web.YassonTranslationTestServlet;
+
+@MinimumJavaLevel(javaLevel = 11)
+@RunWith(FATRunner.class)
+public class YassonTranslationTest extends FATServletClient {
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.jsonb.internal_fat/fat/src/io/openliberty/jakarta/jsonb/internal/fat/YassonTranslationTest.java:38]: This test is only run on JSON-B 3.0.  You also fixed this for JSON-B 1.0 and 2.0.  Should this test also be run for JSON-B 1.0 and 2.0 as well?

File: dev/io.openliberty.jakarta.jsonb.internal_fat/fat/src/io/openliberty/jakarta/jsonb/internal/fat/YassonTranslationTest.java (hunk 8 above)
Comment (KyleAure) [reply dev/io.openliberty.jakarta.jsonb.internal_fat/fat/src/io/openliberty/jakarta/jsonb/internal/fat/YassonTranslationTest.java]: Fixed in latest commit.

File: dev/io.openliberty.jakarta.jsonb.internal_fat/fat/src/io/openliberty/jakarta/jsonb/internal/fat/YassonTranslationTest.java (hunk 9 above)
Comment (KyleAure) [reply dev/io.openliberty.jakarta.jsonb.internal_fat/fat/src/io/openliberty/jakarta/jsonb/internal/fat/YassonTranslationTest.java:38]: Took some finessing but I modified this test to repeat for EE8, EE9, and EE10. 
Thanks for the suggestion

**Comment by KyleAure** [comment]
#libby - Recheck headers
#build - rebuild with changes

**Review by jhanders34 — APPROVED** [review]
I am going to approve.  Technically the jsonb 1.0 and 2.0 repeats should be run on Java 8, but this FAT project is configured to run only with Java 11+, but I am ok with that seeing that the TranslationTest is the only one that runs on the older jsonb levels.

**Pull request merged**